sdk/eventhub/azure-eventhubs/api_review.py

```diff
@@ -0,0 +1,230 @@
+
+class EventHubProducerClient(ClientBaseAsync):
+
+    eh_name  # type: str
+
+    def __init__(
+            self,
+            host: str,
+            event_hub_path: str,
+            credential: TokenCredential,
+            **kwargs
+    ) -> None:
+
+    async def __aenter__(self):
+
+    async def __aexit__(self, *args):
+
+    @classmethod
+    def from_connection_string(
+        cls, conn_str: str,
+        *,
+        event_hub_path: str = None,
+        logging_enable: bool = False,
+        http_proxy: dict = None,
+        auth_timeout: float = 60,
+        user_agent: str = None,
+        retry_total: int = 3,
+        transport_type=None) -> EventHubProducerClient:
+
+    async def send(
+        self,
+        event_data: Union[EventData, EventDataBatch, Iterable[EventData]],
+        *,
+        partition_key: Union[str, bytes] = None,
+        partition_id: str = None,
+        timeout: float = None) -> None:
+
+    async def create_batch(self, events: Iterable[EventData] = None, max_size: int = None) -> EventDataBatch:
+
+    async def get_properties(self) -> Dict[str, Any]:
+
+    async def get_partition_ids(self) -> List[str]:
+
+    async def get_partition_properties(self, partition: str) -> Dict[str, str]:
+
+    async def close(self) -> None:
+
+
+class EventHubConsumerClient(ClientBaseAsync):
+
+    eh_name  # type: str
+
+    def __init__(
+            self,
+            host: str,
+            event_hub_path: str,
+            credential: TokenCredential,
+            *
+            partition_manager: PartitionManager,
+            load_balancing_interval: int = 10,
+            logging_enable: bool = False,
+            http_proxy: dict = None,
+            auth_timeout: float = 60,
+            user_agent: str = None,
+            retry_total: int = 3,
+            transport_type=None
+            **kwargs
+    ) -> None:
+
+    async def __aenter__(self):
+
+    async def __aexit__(self, *args):
+
+    @classmethod
+    def from_connection_string(cls, conn_str: str, **kwargs) -> EventHubConsumerClient:
+
+    async def receive(
+            self,
+            on_event: Callable[PartitionContext, EventData],
+            consumer_group: str,
+            *,
+            partition_id: str = None,
+            owner_level: int = None,
+            prefetch: int = 300,
+            track_last_enqueued_event_properties: bool = False,
+            initial_event_position=None,
```

Comment:
Need to discuss the location of setting the `initial_event_position`, on the receive function or during the on_initialize callback.

Reply:
I'd say on the receive function since users would want to specify where they want to start reading in that partition.

Reply:
One question that was raised was how to specify different sequence numbers for different partitions. Given that this is what the checkpointing is supposed to handle, my (possibly naive) assumption is that this is not a too common scenario. I'm good with taking a single value or a dict (partition_id -> sequence number) for Python for that scenario.

Reply:
Keeping at receive function, and ensure supports cross-partition values.